src/identity/apis/auth.ts

```diff
@@ -283,3 +285,21 @@ export const updateDefaultOrgByAccountID = async ({
 
   // success status code is 204; no response body expected.
 }
+
+// fetch user's default account
+export const getDefaultAccount = async (): Promise<UserAccount> => {
+  const response = await getAccounts({})
+
+  if (response.status === 401) {
+    throw new UnauthorizedError(response.data.message)
+  }
+
+  if (response.status === 500) {
+    throw new ServerError(response.data.message)
+  }
+  const {data} = response
+
+  if (Array.isArray(data) && data.length) {
+    return data.find(account => account.isDefault)
+  }
```

Comment:
This is a much nicer implementation!

However, there are still two cases that will result in `undefined` being returned. `Array.isArray(data) && data.length` returning `false` (which will reach the end of the function without returning anything, which will implicitly return `undefined`), and `data.find(account => account.isDefault)` finding no `account` with `isDefault` set to true. Both of these are very unlikely, but they're still possible, so they should probably be accounted for.

I think doing something like this is probably the easiest way to handle it:

```ts
if (Array.isArray(data) && data.length) {
  const defaultAccount = data.find(account => account.isDefault)
  if (defaultAccount) {
    return defaultAccount
  }
}

throw new GenericError('No default account found') // just a suggestion on the error message. If you can think of a better one, by all means, use that!
```

This way, if everything works, we return the default account, and if not, we fall through and throw the `GenericError`